src/Hipace.cpp

```diff
@@ -638,19 +644,35 @@ Hipace::ExplicitSolveOneSubSlice (const int lev, const int ibox, const amrex::Bo
     m_grid_current.DepositCurrentSlice(m_fields, geom[lev], lev, islice);
     // No FillBoundary because grid current only deposits in the middle of the field
 
-    // Modifies Bx, By, Sx and Sy in the current slice
-    ExplicitSolveBxBy(lev);
+    // Set Sx and Sy to beam contribution
+    InitializeSxSyWithBeam(lev);
+
+    // Deposit Sx and Sy for every plasma species
+    m_multi_plasma.ExplicitDeposition(m_fields, m_laser, geom[lev], lev);
+
+    // Solves Bx, By using Sx, Sy and chi
+    ExplicitMGSolveBxBy(lev, WhichSlice::This);
+
+    const bool do_salame = m_multi_beam.isSalameNow(step, islice_local, beam_bin);
+    if (do_salame) {
+        SalameModule(this, m_salame_n_iter, m_salame_do_advance, m_salame_last_slice,
```

Comment:
```suggestion
        // Modify the beam particle weights on this slice to flatten Ez.
        // As the beam current is modified, Bx, By and Ez are also recomputed,
        // as well as the plasma particle.
        SalameModule(this, m_salame_n_iter, m_salame_do_advance, m_salame_last_slice,
```
Is this correct? Is there something else that is changed and should be specified here?
Also, when `do_beam_jz_minus_rho`, doesn't SALAME also modify rho, affecting `ExmBy` and `EypBx`? Likewise, doesn't SALAME affect `Bz`, and shouldn't it be recomputed afterwards? If that's an acceptable approximation, totally fine, but we could just write a sentence somewhere in the doc or so.

Reply:
As stated in the documentation for `do_salame`, the salame beam contribution to jx, jy, and rhm-jz (but not jz) will be ignored in the first time-step, so Psi, Ex and Bz do not depend on the salame weight.